src/proof_rules/induction.rs

```diff
@@ -192,6 +207,18 @@ impl Encoding for KIndAnnotation {
         resolve.visit_expr(invariant)
     }
 
+    fn check_calculus(&self, calculus: &Calculus, direction: Direction) -> Result<(), ()> {
```

Comment:
This method should return a descriptive error type, e.g. just an `AnnotationError`